### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by m-fedorov-s**
Written test for geometry module, optimized structure of the repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,4 +4,10 @@ Use `cargo` to build and run this project.
...
 
-    $ curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
\ No newline at end of file
+    $ curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
+
+You may run tests by
+
+    $ cargo test
+
+Don' t forget run `cargo fmt` before creating merge request.
```
Comment (Gleefre) [inline README.md]: Typos: `don' t` -> `don't`; `forget run` -> `forget to run`.

File: src/geometry.rs [hunk 2]
```diff
@@ -0,0 +1,180 @@
+#![allow(unused)]
+#[derive(Debug)]
+pub struct Point(f64, f64, f64);
+struct Vector(f64, f64, f64);
```
Comment (Gleefre) [inline src/geometry.rs]: Why vector is not a public structure?

File: src/geometry.rs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    #[test]
+    fn shift_by_zero_vector() {
+        let point = Point(1., 5., 6.);
+        let vector = Vector(0., 0., 0.);
+        assert_eq!(point.0, point.shift_by(&vector).0);
+        assert_eq!(point.1, point.shift_by(&vector).1);
+        assert_eq!(point.2, point.shift_by(&vector).2);
+        assert_eq!(0., vector.norm());
```
Comment (Gleefre) [inline src/geometry.rs]: This assertion doesn't correspond to test's name. Maybe it should be moved to `normalizing vector` test?

File: src/geometry.rs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+// Кроме того, чтобы поддерживать текстуры необходимо рассчитывать барацентрические
+// координаты точки в треугольнике, а для каждой вершины треугольника помнить координаты
+// этой верщины в текстуре. Кроме того, для улучшения качества нужно помнить нормали в
+// каждой вершине, и рассчитывать нормаль в точке как взвешенную сумму нормалей в
+// вершинах, с весами, равными барацентрическим координатам.
+
+// struct Sphere {...}
+// Хотим поддержку шариков ведь, так?)
+
+#[cfg(test)]
```
Comment (Gleefre) [inline src/geometry.rs:93]: Why not to separate geometry module with geometry test module?

File: src/geometry.rs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        assert_eq!(a_point.distance(&b_point), 6.);
+    }
+
+    #[test]
+    fn shift_by_zero_vector() {
+        let point = Point(1., 5., 6.);
+        let vector = Vector(0., 0., 0.);
+        assert_eq!(point.0, point.shift_by(&vector).0);
+        assert_eq!(point.1, point.shift_by(&vector).1);
+        assert_eq!(point.2, point.shift_by(&vector).2);
```
Comment (Gleefre) [inline src/geometry.rs]: Is it possible to write custom equality check for point to avoid code duplication?

File: src/geometry.rs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    #[test]
+    fn vector_product_for_standard_basis() {
+        let a_vector = Vector(1., 0., 0.);
+        let b_vector = Vector(0., 1., 0.);
+        let product = a_vector.vector_product(&b_vector);
+        assert_eq!(0., product.0);
+        assert_eq!(0., product.1);
+        assert_eq!(1., product.2);
```
Comment (Gleefre) [inline src/geometry.rs]: Is it possible to write custom equality check for vector to avoid code duplication?

**Review by Gleefre — COMMENTED** [review]
Maybe `main.rs` and `scene.rs` should not be merged.

File: src/geometry.rs (hunk 4 above)
Comment (m-fedorov-s) [reply src/geometry.rs:93]: It is a common practice, according to the rust book.
"The convention is to create a module named tests in each file to contain the test functions and to annotate the module with cfg(test)." See [this link](https://doc.rust-lang.org/book/ch11-03-test-organization.html)

Comment (Gleefre) [reply src/geometry.rs:93]: Ok.

File: src/geometry.rs [hunk 7]
```diff
@@ -133,10 +141,7 @@ mod tests {
     fn shift_by_zero_vector() {
         let point = Point(1., 5., 6.);
         let vector = Vector(0., 0., 0.);
-        assert_eq!(point.0, point.shift_by(&vector).0);
-        assert_eq!(point.1, point.shift_by(&vector).1);
-        assert_eq!(point.2, point.shift_by(&vector).2);
-        assert_eq!(0., vector.norm());
```
Comment (Gleefre) [inline src/geometry.rs]: This test has been completely removed. Maybe it should be added back in a separate test function?

**Review by Gleefre — APPROVED** [review]

**Pull request merged**